### ISSUE TITLE
Add more information to stock `datakit project create` error

### ISSUE THREAD

**Issue by myersjustinc** [associatedpress/datakit-project#7]
@kastanis noted yesterday that on a new machine that hasn't used `datakit project create` before, the resulting error message doesn't provide enough information by itself to help the user move forward:

```
$ datakit project create
No project templates have been installed. You must specify the local path or URL to a Cookiecutter project repo.
```

At the very least, we should explain that there is a `--template` flag that is required the first time so datakit-project will know what kind of project you want to work on by default, and we should provide (or link to) a list of common templates so the user has some sense of what to do next.

It also might be possible to prompt interactively, especially if we have a list of potential templates.

**Comment by myersjustinc** [comment]
In the meantime, most AP data analysis staffers should use the following when running into this particular error:

```
$ datakit project create --template https://github.com/associatedpress/cookiecutter-r-project
```

**Comment by tthibo** [comment]
I really like the idea of eventually handling this interactively. We'll likely need that functionality to implement the default Gitlab ticket(s) we've discussed (we'll need different default tickets if there is a distro contemplated for the project, for example.)

**Comment by myersjustinc** [comment]
My concern with handling this particular issue interactively is that we'd need to come up with (and maintain) the list of templates. Since Cookiecutter can pull a template from literally anywhere, we'll need an "Other" option at the very least, but then how do we solicit and/or vet templates to include in this list?

It's doable, but it's one of those situations where the social and procedural aspects are much thornier than the technical aspects.

**Comment by tthibo** [comment]
Wondering if there's a configuration option we should explore here that lets users select a project type, which would include some sensible defaults defined in a JSON. I'm thinking primarily about the ways we work now, and it might be nice to do something like tell datakit you're doing an R analysis project, which would select our default R project template and choose a default starting Gitlab ticket. Choosing that project type could trigger a second question "Is there a data distribution planned?' which would add other elements to the project and a distribution checklist gitlab ticket. 

Seems like having some configurability there would allow for a few quick project customizations based on some standard categories. Might be difficult to generalize to something useful for other teams, though. 

Could be overkill. I'd want to hear from the other folks doing analysis work whether this sort of thing seems worth the trouble and what (if any) project definition decisions might be useful as part of an interactive approach.

**Comment by zstumgoren** [comment]
+1 to improve the error message on the existing `project create` command. Seems like a sensible and easy update.

But yeah, an interactive install from remote resources could get tricky since there's no central repository of templates (that I'm aware of) akin to packages on PyPI. The cookiecutter docs have a [pantry full of templates](https://cookiecutter.readthedocs.io/en/latest/readme.html#a-pantry-full-of-cookiecutters), but I'd expect anyone interested in these would first want to review a template before installing to ensure it's appropriate for the use case. A very common workflow, for example, would be to fork and modify an existing template on github before installing it.

But giving users the ability to easily choose between *locally installed templates* during project creation seems like a worthwhile goal. Dkit already has the notion of a default project type. And a [command to list locally installed templates](https://github.com/associatedpress/datakit-project/issues/2) has long been on the To Do list. If we built the "list" command, shouldn't be too much extra trouble to integrate the template discovery that would go along with it into an interactive mode for creating a new project.

**Comment by zstumgoren** [comment]
@tthibo's ideas about some of the more AP-specific workflows are good ones as well, though I'd agree as he points out that some of them might be hard to generalize for other users. Perhaps it's worth building out the basic functionality for template listing and interactive project creation into the datakit-project plugin, and then update the closed-source datakit-ap to include the additional behaviors as part of some higher-level commands? These higher-level commands could easily import the more general code we write for the open-source project plugin. If we find that features for datakit-ap are in fact generally useful, it's easy enough to port those over to the core `project` plugin.

**Comment by zstumgoren** [comment]
@myersjustinc @tthibo Think we should migrate the conversations about interactive mode and general-purpose installation scripts to new tickets.

Apropos the original issue (i.e. a better error message when there are no templates installed), what do you all and @kastanis think about the below help text?

```
No project templates have been installed!
You must specify the local path or URL to a Cookiecutter project repo, e.g.:

    datakit project create --template /local/path/to/my-awesome-cookiecutter-template

Check out the below URLs for many more pre-baked templates and more installation options:

  - A Pantry Full of Cookiecutters
      https://cookiecutter.readthedocs.io/en/latest/readme.html#a-pantry-full-of-cookiecutters

  - Template install docs
      https://datakit-project.readthedocs.io/en/latest/#creating-your-first-project

```

**Comment by myersjustinc** [comment]
This sounds reasonable to me, including the suggested ticket migration.

**Comment by zstumgoren** [comment]
This is implemented on v0.3.0 branch. Details on implementation and new help text are in https://github.com/associatedpress/datakit-project/issues/10#issuecomment-504170617
